simulator/src/main/java/Environment.java

```diff
@@ -1,36 +1,94 @@
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 import java.util.concurrent.ConcurrentLinkedQueue;
 
 public class Environment
 {
-
-    private final Map<Integer, ConcurrentLinkedQueue<byte[]>> packets_in_transit;
+    int num_nodes;
+    private Map<Integer, ConcurrentLinkedQueue<Message>> packets_in_transit;
+    private Map<Integer, Map<Integer, Integer>> signal_strengths;
 
     public Environment()
     {
-        this.packets_in_transit = Collections.synchronizedMap(new HashMap<Integer, ConcurrentLinkedQueue <byte[]>>());
+        this.num_nodes = 4;
```

Comment:
Should this be a command line argument when running the simulator?

Reply:
Yeah it should be i think thats more of a nice to have at this point... this will work rn for routing testing

Reply:
This could also be a improvement labeled github issue